### PR TITLE
Fix etcd lock error

### PR DESCRIPTION
**Pull request by khushboobhatia01**
Fixes https://github.com/StackStorm/st2/issues/5483

### CODE REVIEW

**Review by armab — COMMENTED** [review]
Thanks, @khushboobhatia01 for the fix! :+1: 

Could you include a Changelog record for this regression bugfix?

File: CHANGELOG.rst [hunk 1]
```diff
@@ -11,6 +11,10 @@ in development
 Added
 ~~~~~
 
+* Use string lock name instead of object id. #5484
+
+  Contributed by @khushboobhatia01
+
```
Comment (armab) [inline CHANGELOG.rst:17]: The changelog would need to go into `development` section as `v3.6.0` was already cut.
The record best fit for the `Bugfix` section. Please also mention that's it's a fix for a regression, referencing the #5358.
This way we'll keep the issues connected.

**Review by armab — APPROVED** [review]
Thanks for the hotfix! :+1:

**Review by cognifloyd — APPROVED** [review]
Thanks for fixing the Changelog!
The fix makes sense.

**Pull request merged**